### ISSUE TITLE
Rule Change: only allow object as a return value instead of `super()` in `constructor-super`

### ISSUE THREAD

**Issue by Tanujkanti4441** [eslint/eslint#21017]
### What rule do you want to change?

`constructor-super`

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
class A extends B { constructor() { return 1; } }
```

### What does the rule currently do for this code?

Rule does not report this and assumes `return` value as a substitute of 'super()' but this code gives error as it should only return an object instead.
`TypeError: Derived constructors may only return object or undefined`

https://github.com/eslint/eslint/blob/bb9eb2ab28daf6e6dafc62daa6b5daf9c2b7046c/tests/lib/rules/constructor-super.js#L67-L70

### What will the rule do after it's changed?

Rule will report if a constructor returns a value, except an object or identifier, instead of super().

### Participation

- [x] I am willing to submit a pull request to implement this change.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

**Comment by DMartens** [comment]
ESLint provides the rule [no-constructor-return](https://eslint.org/docs/latest/rules/no-constructor-return). It disallows any return with an explicit value rather than allowing only subset as proposed here. As such I think we should discuss this as adding an option to `no-constructor-return`.

**Comment by mdjermanovic** [comment]
I don't think this check belongs in the `constructor-super` rule, because returning a primitive value from a constructor is an error on its own,  unrelated to `super()` calls. I think this check should rather be a separate rule.

Since we don't have a consensus on this change, I'm closing the issue.
